packages/core/src/hooks/useEmptyValueOnChange.js

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useEffect } from 'react';
+
+const useEmptyValueOnChange = ({ onChange, options, value }) => {
+  const mergeValues = useCallback((newValue) => {
+    return newValue === "" && options?.emptyValue ? options?.emptyValue : newValue;
+  }, [options]);
+
+  const rawOnChange = useCallback((newValue) => {
+    return onChange(mergeValues(newValue));
+  }, [onChange, mergeValues]);
+
+  const eventOnChange = useCallback((event) => {
+    const newValue = event?.target?.value;
+    return rawOnChange(newValue);
+  }, [rawOnChange]);
+
+  useEffect(() => {
+    if (value === mergeValues(value)){
+      return;
```

Comment:
Added in this guard clause because I was seeing tests fail for `onChange` getting called multiple times if the value wasn't empty.